### PR TITLE
Added Tests for Wifi Server and Error Status code for Error Response

### PR DESCRIPTION
**Pull request by thienhoang23**

### CODE REVIEW

File: cmds/wifi/wifi.go [hunk 1]
```diff
@@ -73,6 +73,10 @@ func init() {
 }
 
 func scanWifi() error {
+	if *test {
```
Comment (rminnich) [inline cmds/wifi/wifi.go]: hmm. could you just have a function called doNothing, and a function called doScanWiFi, and the have this
scanWiFi = doScanWiFi
and then
if *test {
scanWiFi = doNothing
}
?

This way of doing it just seems weird to me

File: cmds/wifi/wifi.go [hunk 2]
```diff
@@ -210,6 +214,10 @@ func connectWifiArbitrator() {
 }
 
 func connectWifi(a ...string) error {
+	if *test {
```
Comment (rminnich) [inline cmds/wifi/wifi.go]: same here

File: cmds/wifi/wifi.go [hunk 3]
```diff
@@ -286,6 +294,8 @@ func main() {
 		return
 	}
 
+	// Start a Server with Stub data
+	// for manual end-to-end testing
```
Comment (rminnich) [inline cmds/wifi/wifi.go]: This just seems like you need two functions and the pointer suggested above?
Or ... make an interface with Scan and Connect and then have a Null implementation of Scan and Connect

This just doesn't seem right.

File: cmds/wifi/wifi.go (hunk 1 above)
Comment (thienhoang23) [reply cmds/wifi/wifi.go]: Fixed

File: cmds/wifi/wifi.go (hunk 2 above)
Comment (thienhoang23) [reply cmds/wifi/wifi.go]: Fixed

File: cmds/wifi/wifi.go (hunk 3 above)
Comment (thienhoang23) [reply cmds/wifi/wifi.go]: Fixed

File: cmds/wifi/wifi.go [hunk 4]
```diff
@@ -8,29 +8,16 @@ import (
...
-	"io/ioutil"
 	"log"
 	"os"
 	"os/exec"
-	"regexp"
-	"strings"
 
-	"github.com/u-root/u-root/pkg/wpa/passphrase"
+	"github.com/u-root/u-root/pkg/wifi"
+	"github.com/u-root/u-root/pkg/wifi/wifiStub"
```
Comment (rminnich) [inline cmds/wifi/wifi.go]: just make it part of the package. If it's not used it won't get compiled inl

File: pkg/wifi/wifiStub/wifiStub.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+	return stub.ScanWifiOut, nil
+}
+
+func (stub StubWifiWorker) ScanCurrentWifi() (string, error) {
+	return stub.ScanCurrentWifiOut, nil
+}
+
+func (StubWifiWorker) Connect(a ...string) error {
+	return nil
+}
```
Comment (rminnich) [inline pkg/wifi/wifiStub/wifiStub.go]: I'm going to assume the elements in StubWifiWorker are never set, which means you ought to be able to do this:
type NullWifi int
func (NullWifi) ScanInterfaces ([]string, error) {
return []string{}, nil
}

and so on
Then when you use this you can do something like
var v wifi.NullWifi
and use v.ScanInterfaces

and so on

and go ahead and not have this a separate package.

File: cmds/wifi/wifi.go (hunk 4 above)
Comment (thienhoang23) [reply cmds/wifi/wifi.go]: Fixed

**Pull request merged**